.github/workflows/compare_index_changes.py

```diff
@@ -24,31 +25,66 @@ def compare_index_changes(new_index, old_index):
     old_modules = old_index["index"]
 
     for module_name in new_modules:
-        if module_name not in old_modules:
-            continue
-
         new_module = new_modules[module_name]
-        old_module = old_modules[module_name]
-
-        # Handled by another GH Action
-        if "version" not in new_module:
-            continue
-        if "commit" not in new_module:
-            continue
-        if "version" not in old_module:
-            continue
-        if "commit" not in old_module:
-            continue
-
-        if new_module["version"] != old_module["version"]:
-            if new_module["commit"] == old_module["commit"]:
-                print("Error: Attribute 'version' was changed, but not 'commit'; in module '%s'" % module_name)
-                sys.exit(1)
-
-        if new_module["commit"] != old_module["commit"]:
-            if new_module["version"] == old_module["version"]:
-                print("Error: Attribute 'commit' was changed but not 'version'; in module '%s'" % module_name)
-                sys.exit(1)
+        old_module = old_modules.get(module_name)
+
+        # Any remote resources must be reachable
+        validate_reachable_resources(new_module, old_module, module_name)
+
+        if old_modules:
+            # If either version or commit is changed, then both must be changed
+            validate_version_and_commit(new_module, old_module, module_name)
+
+
+def validate_version_and_commit(new_module, old_module, module_name):
+    if new_module.get("version") != old_module.get("version"):
+        if new_module.get("commit") == old_module.get("commit"):
+            print(
+                "Error: Attribute 'version' was changed, but not 'commit'; in module '%s'"
+                % module_name
+            )
+            sys.exit(1)
+
+    if new_module.get("commit") != old_module.get("commit"):
+        if new_module.get("version") == old_module.get("version"):
+            print(
+                "Error: Attribute 'commit' was changed but not 'version'; in module '%s'"
+                % module_name
+            )
+            sys.exit(1)
+
+
+def validate_reachable_resources(new_module, old_module, module_name):
+    # To save bandwidth we only HEAD request on changes to an URL
+    def _head(url):
+        response = requests.head(url)
+        if not response.ok:
+            print(
+                f"Error: remote resource '{url}' in module '{module_name}' is not reachable"
```

Comment:
Don't use `f""` strings, please.

Reply:
Does this really matter in a GHA ?

Reply:
Right, I thought this was part of *cfbs*. Sorry for the noise.